src/runtime/lib/telemetry/index.ts

```diff
@@ -85,62 +103,112 @@ export class TelemetryClient {
         newUser: boolean;
         timeUploaded: Date | null;
         clientId: string;
+        operatingSystem: string | null;
+        dependencyVersion: string | null;
+        gitVersion: string | null;
         actions: MetricAction[];
     };
     private postRetries: number;
     private serverUrl: string;
     private enabled: boolean;
     private httpClient: AxiosInstance;
+
+    private static instance: TelemetryClient;
+
     /**
      * @constructor
      * @param {TelemetryClientOptions} options - The options for constructor
      */
-    constructor(options: TelemetryClientOptions) {
-        const { } = options;
+    private constructor(options?: TelemetryClientOptions) {
+        if (options) {
+            const { } = options;
+        }
         this.httpClient = axios.create({ timeout: 3000, headers: { 'Content-Type': 'text/plain' } });
         this.serverUrl = 'https://client-telemetry.amazonalexa.com';
         this.postRetries = 3;
         this.enabled = this.isEnabled();
+        const osType = os.type();
+        const osArch = os.arch();
+        const osRelease = os.release();
+        const gitVersionUint8Array: Uint8Array = child_process.execSync(
+            'git --version');
+        const gitVersion = new TextDecoder(DEFAULT_ENCODING).decode(gitVersionUint8Array);
```

Comment:
Will this lead to an error if `git` is not installed or is not on `$PATH` ? I think it is better to put this under a `try-catch` block

Reply:
Added a try catch